extensions/ql-vscode/src/common/commands.ts

```diff
@@ -0,0 +1,24 @@
+import { CommandManager } from "../packages/commands";
+
+/**
+ * Contains type definitions for all commands used by the extension.
+ *
+ * To add a new command first define its type here, then provide
+ * the implementation in the corresponding `getCommands` function.
+ */
+
+// Commands directly in the extension
+export type ExtensionCommands = {
+  "codeQL.openDocumentation": () => Promise<void>;
+};
+
+// Commands tied to variant analysis
+export type VariantAnalysisCommands = {
+  "codeQL.openVariantAnalysisLogs": (
+    variantAnalysisId: number,
+  ) => Promise<void>;
+};
+
+export type AllCommands = ExtensionCommands & VariantAnalysisCommands;
+
+export type ExtensionCommandManager = CommandManager<AllCommands>;
```

Comment:
I think I prefer `App` over `Extension` as well, do you have any thoughts on just renaming that @koesie10 ?

Reply:
I think naming it `App` makes a lot of sense. I'm not sure about renaming `ExtensionCommands` to `AppCommands` though since `ExtensionCommands` doesn't actually contain all commands of the app, it only contains the commands registered in the `extension.ts` file. I think having `AllCommands` as `AppCommands` would make more sense, but I'll leave that up to @starcke to decide which one he prefers.

Reply:
`AppCommandManager` is good, but thinking about it I think both `ExtensionCommands` and `AppCommands` are a bit wrong as `VariantAnalysisCommands` are also commands for the extension/app. So I think something like `BaseCommands` for the commands that are not defined in any specific manager (like the variant analysis manager).

Reply:
Thanks both!